### DISCUSSION TITLE
[Feedback v2.0.0 (beta1) ] - performTextFinderAction feature seems broken

### DISCUSSION THREAD

**Discussion by paul1893** [krzyzanowskim/STTextView#74] [Questions]
Hello 👋,

If I'm not mistaken, `STLineNumberRulerView` has been replaced by a `STGutterView` view in **v2.0.0 (beta 1)**. It is no longer a `NSRulerView` and so `performTextFinderAction` feature seems broken ? As `NSPanel` and position of search results are now wrong offset.

<img width="1116" alt="Capture d’écran 2024-11-19 à 22 43 23" src="https://github.com/user-attachments/assets/8005e50b-2c6a-4d2b-a5bb-35974c9f67d3">

https://github.com/user-attachments/assets/f97939b3-28c8-4c9b-97e5-fde7a7835d1e

**Comment by krzyzanowskim** [comment]
yes. It's the first thing to fix when I sit on it. unless you want to get a try on that?

**Reply by paul1893** [reply]
If you have it in mind, that's the main thing. 
I'm not sure I'll find the time in the coming weeks to take a detailed look, but if I do, would you prefer me to open a PR?

To ensure we're synced about resolution strategy, would you prefer that STGutterView inherits from NSRulerView? (This is what I have in mind) or keeping STGutterView a standard NSView (I guess you made it for a reason) ?

**Reply by paul1893** [reply]
I've quickly jump on the subject this evening and got something that seems to work when making `STGutterView` a `NSRulerView` again 🤷‍♂️

https://github.com/user-attachments/assets/26514eba-55f8-4253-9e4c-eae497671480

**Comment by krzyzanowskim** [comment]
> would you prefer that STGutterView inherits from NSRulerView?

I'd rather not. for many reasons, but dealing with NSRuleView turned out to be too limited - also not transferable to UIKit

**Reply by paul1893** [reply]
>  I'd rather not.  ... also not transferable to UIKit

Ok, understood 👌

**Reply by paul1893** [reply]
Ok, so adding `scrollView.contentInsets.left = gutterPadding` in the `STTextView.sizeToFit` method and setting `scrollView.automaticallyAdjustsContentInsets = false` to avoid weird jumps when resizing the window seems to makes the job and let the `STGutterView` being a `NSView` untouched.

<img width="800" alt="Capture d’écran 2024-11-22 à 09 42 02" src="https://github.com/user-attachments/assets/6ed89bac-9a00-4d9a-b896-a73a1516b6ae">


https://github.com/user-attachments/assets/f4bd3cdc-e745-4832-bb63-e27efe98370b

**Reply by krzyzanowskim** [reply]
that's the thing! want to open PR with that change?

**Reply by paul1893** [reply]
Yes sure, here it is: https://github.com/krzyzanowskim/STTextView/pull/75

Don't hesitate to test it deeper. I'm not sure to know all about STTextView and side effect that I could have been introduced.

**Comment by krzyzanowskim** [comment]
Fixed on `main` branch now.

https://github.com/user-attachments/assets/598aec03-3ab0-4940-a758-49463601aa6d
